lib/galaxy/webapps/galaxy/controllers/history.py

```diff
@@ -1057,7 +1063,12 @@ def purge_deleted_datasets(self, trans):
                 if not hda.deleted or hda.purged:
                     continue
                 if trans.user:
-                    trans.user.adjust_total_disk_usage(-hda.quota_amount(trans.user))
+                    if not hda.dataset.has_active_storage_media():
+                        trans.user.adjust_total_disk_usage(-hda.quota_amount(trans.user))
```

Comment:
I think we can get around duplicating this code in the controllers with https://github.com/galaxyproject/galaxy/pull/10208.

Reply:
That other PR has been merged so I think this should be rebased now along with the fix for https://github.com/galaxyproject/galaxy/pull/4840/files#r486551671.